dashboard/src/components/DeploymentFormBody/DeploymentFormBody.test.tsx

```diff
@@ -382,3 +382,106 @@ it("restores the default chart values when clicking on the button", () => {
 
   expect(setValues).toHaveBeenCalledWith("foo: value");
 });
+
+[
+  {
+    description: "should merge modifications from the values and the new version defaults",
+    defaultValues: "foo: bar\n",
+    deployedValues: "foo: bar\nmy: var\n",
+    newDefaultValues: "notFoo: bar",
+    result: "notFoo: bar\nmy: var\n",
+  },
+  {
+    description: "should modify the default values",
+    defaultValues: "foo: bar\n",
+    deployedValues: "foo: BAR\nmy: var\n",
+    newDefaultValues: "foo: bar",
+    result: "foo: BAR\nmy: var\n",
+  },
+  {
+    description: "should delete an element in the defaults",
+    defaultValues: "foo: bar\n",
+    deployedValues: "my: var\n",
+    newDefaultValues: "foo: bar\n",
+    result: "my: var\n",
+  },
+  {
+    description: "should add an element in an array",
+    defaultValues: `foo:
+  - foo1:
+    bar1: value1
+`,
+    deployedValues: `foo:
+  - foo1: 
+    bar1: value1
+  - foo2: 
+    bar2: value2
+`,
+    newDefaultValues: `foo:
+    - foo1:
+      bar1: value1
+`,
+    result: `foo:
+  - foo1: 
+    bar1: value1
+  - foo2: 
+    bar2: value2
+`,
+  },
+  {
+    description: "should delete an element in an array",
+    defaultValues: `foo:
+  - foo1:
+    bar1: value1
+  - foo2:
+    bar2: value2
+`,
+    deployedValues: `foo:
+  - foo2: 
+    bar2: value2
+`,
+    newDefaultValues: `foo:
+  - foo1:
+    bar1: value1
+  - foo2:
+    bar2: value2
+`,
+    result: `foo:
+  - foo2: 
+    bar2: value2
+`,
+  },
+].forEach(t => {
+  it(t.description, () => {
+    const selected = {
+      ...defaultProps.selected,
+      versions: [chartVersion],
+      version: chartVersion,
+      values: t.defaultValues,
+      schema: initialSchema,
+    };
+    const newSelected = {
+      ...defaultProps.selected,
+      versions: [chartVersion],
+      version: chartVersion,
+      values: t.newDefaultValues,
+      schema: initialSchema,
+    };
+    const setValues = jest.fn();
+    const wrapper = shallow(
+      <DeploymentFormBody
+        {...props}
+        deployedValues={t.deployedValues}
+        setValues={setValues}
+        selected={selected}
+      />,
+    );
+    // Store the modifications
+    wrapper.setProps({ selected: props.selected });
```

Comment:
sorry, these two lines are actually unnecessary since the DeploymentFormBody initialization with the `selected` chart there already executes `componentWillReceiveProps`.

The issue is that with this code is really being executed, the `selected` property is undefined, it's not until the component executes the `componentDidMount` and therefore `getChartVersion` when the `selected` chart is requested and eventually populated.

Reply:
Sorry, it was actually needed (I was executing the wrong tests). I have adapted them to reflect reality in the most similar way... But maybe you are right, we may need to request the default chart version before and pass it to this component even though I would prefer to do so in a different PR. Let me know what you think.